app/src/main/java/org/simple/clinic/util/IdentifierDisplayAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package org.simple.clinic.util
+
+import org.simple.clinic.patient.businessid.Identifier
+import org.simple.clinic.patient.shortcode.UuidShortCode
+import org.simple.clinic.patient.shortcode.UuidShortCodeCreator
+import java.util.UUID
+import javax.inject.Inject
+
+/**
+ * We need to render items of [Identifier] type in the UI, sometimes differently from how
+ * they are represented in code.
+ *
+ * For example, the short code of a [Identifier.IdentifierType.BpPassport] is represented as a
+ * 7-digit string, ex ("1234567"), but when rendering it in the UI we need to display it as
+ * "123 4567". We do not want to store these strings with the spaces in business layer because this
+ * is a presentation detail.
+ *
+ * This class is a helper class that is meant to convert [Identifier] instances to their display
+ * formats.
+ **/
+class IdentifierDisplayAdapter @Inject constructor(private val uuidShortCodeCreator: UuidShortCodeCreator) {
+
+  fun toCharSequence(identifier: Identifier): CharSequence {
+    return when (identifier.type) {
+      Identifier.IdentifierType.BpPassport -> bpPassportToCharSequence(identifier.value)
```

Comment:
Creating a method for each type of `Identifier` will quickly lead to one method for each type.
Eventually the churn of this file would be more.

I think a better way to implement this would be to create separate adapters for each `Identifier` and then create a map of `IdentifierType` to adapter.
We can then inject this map to the screens that need the identifier display.

Reply:
Also, I just realized that we are injecting `UuidShortCodeCreator` which is specific to `BpPassport` whereas this class is common to all `Identifier`s

Reply:
Sounds good. Done.